app/controllers/reviewable_notes_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class ReviewableNotesController < ApplicationController
+  requires_login
+  before_action :ensure_staff
+  before_action :find_reviewable
+
+  def create
+    note = @reviewable.reviewable_notes.build(note_params)
+    note.user = current_user
+
+    if note.save
+      # Reload to ensure associations are loaded
+      note.reload
+      render json: ReviewableNoteSerializer.new(note, scope: guardian, root: false)
+    else
+      render json: { errors: note.errors.full_messages }, status: 422
+    end
+  end
+
+  def destroy
+    note = @reviewable.reviewable_notes.find(params[:id])
+
+    # Only allow the author or admin to delete notes
+    raise Discourse::InvalidAccess unless note.user == current_user || current_user.admin?
+
+    note.destroy
```

Comment:
```suggestion
    note.destroy!
```

`destroy` can return `false` if the query to delete the record fails for some reason. It is better to use `destroy!` to raise an error on failure since we are not checking for the return value.